### PR TITLE
Add option to perform Euclidean cluster extraction in 2D (#3084)

### PR DESCRIPTION
**Pull request by chambbj**
Changed to return clusters as deque of PointIdLists (rather than
vector). Default behavior will continue to be to cluster in 3D.

### CODE REVIEW

**Comment by chambbj** [comment]
Yes, it is. And you are right, it could be simplified.

**Comment by chambbj** [comment]
It occurs to me that this needs a slight doc tweak (and a test or two for 2D clusters).

File: filters/private/Segmentation.hpp [hunk 1]
```diff
@@ -91,10 +91,68 @@ std::ostream& operator<<(std::ostream& out, const PointClasses& classes);
   \param[in] tolerance the tolerance for adding points to a cluster.
   \returns a vector of clusters (themselves vectors of PointIds).
```
Comment (abellgithub) [inline filters/private/Segmentation.hpp:92]: No longer vectors.

In this case, I can't imagine it much matters if you use deque or vector.

File: filters/private/Segmentation.hpp [hunk 2]
```diff
@@ -91,10 +91,68 @@ std::ostream& operator<<(std::ostream& out, const PointClasses& classes);
...
 */
-PDAL_DLL std::vector<PointIdList> extractClusters(PointView& view,
-                                                  uint64_t min_points,
-                                                  uint64_t max_points,
-                                                  double tolerance);
+template <class T>
```
Comment (abellgithub) [inline filters/private/Segmentation.hpp:94]: T might be named better in this case, like KDINDEX or something.

**Pull request merged**